assets/js/components/KeyMetrics/MetricsSelectionPanel/Metrics.js

```diff
@@ -32,19 +32,21 @@ import MetricItem from './MetricItem';
 const { useSelect } = Data;
 
 export default function Metrics() {
-	const availableMetrics = useSelect( ( select ) =>
-		pickBy( KEY_METRICS_WIDGETS, ( _value, key ) => {
+	const availableMetrics = useSelect( ( select ) => {
+		const { isModuleConnected } = select( CORE_MODULES );
+
+		return pickBy( KEY_METRICS_WIDGETS, ( _value, key ) => {
 			const widget = select( CORE_WIDGETS ).getWidget( key );
```

Comment:
The same optimization we did with `isModuleConnected` applies here as well.

Reply:
Ah, excellent point, thank you for spotting this!